src/Utilities/product_of_sets.jl

```diff
@@ -103,26 +109,26 @@ function MOI.get(
         S in set_types(sets) if set_index(sets, S) in present
     ]
 end
+
 function MOI.get(
     sets::MixOfScalarSets,
     ::MOI.NumberOfConstraints{F,S},
 ) where {F,S}
     i = set_index(sets, S)
     return count(isequal(i), sets.set_ids)
 end
+
 function MOI.get(
     sets::MixOfScalarSets,
     ::MOI.ListOfConstraintIndices{F,S},
 ) where {F,S}
     i = set_index(sets, S)
-    return LazyMap{MOI.ConstraintIndex{F,S}}(
-        j -> MOI.ConstraintIndex{F,S}(j),
-        Base.Iterators.Filter(
-            j -> sets.set_ids[j] == i,
-            eachindex(sets.set_ids),
-        ),
-    )
+    return MOI.ConstraintIndex{F,S}[
```

Comment:
What's the point of collecting ? Being able to iterate through the list of indices without needing to collect them can be useful

Reply:
Simplifying the number of types we are using. Unless we identify something as a bottleneck, we should use the simplest approach. Adding the LazyMap here just complicates things.

Reply:
Simple is a bit subjective. `LazyMap` is already used a lot in bridges and so on so it's not like we introduce another type.
Writing things slower until we have a benchmark isn't the right approach either. We should at least write it the fastest way if it does not decrease readability too much. In this case, I don't think it affects readability that much.

Reply:
We don't have any benchmarks showing that `LazyMap` is faster.

Reply:
Repeating my point below, we can return to this once everything is hooked up and we can run benchmarks on solvers. Until then, I'm leaving as-is.